src/java.base/share/classes/sun/security/ssl/SSLTransport.java

```diff
@@ -122,16 +123,35 @@ static Plaintext decode(TransportContext context,
         } catch (AEADBadTagException bte) {
             throw context.fatal(Alert.BAD_RECORD_MAC, bte);
         } catch (BadPaddingException bpe) {
-            /*
-             * The basic SSLv3 record protection involves (optional)
-             * encryption for privacy, and an integrity check ensuring
-             * data origin authentication.  We do them both here, and
-             * throw a fatal alert if the integrity check fails.
-             */
-             Alert alert = (context.handshakeContext != null) ?
-                     Alert.HANDSHAKE_FAILURE :
-                     Alert.BAD_RECORD_MAC;
-            throw context.fatal(alert, bpe);
+            // Check for unexpected plaintext alert message during TLSv1.3 handshake, @bug 8331682
+            if (srcsLength == 1 && !context.isNegotiated && context.handshakeContext != null &&
+                    ProtocolVersion.TLS13.equals(context.handshakeContext.negotiatedProtocol)) {
+                ByteBuffer packet = srcs[srcsOffset].duplicate();
+                packet.position(0);
+                byte contentType = packet.get();                   // pos: 0
+                byte majorVersion = packet.get();                  // pos: 1
+                byte minorVersion = packet.get();                  // pos: 2
+                int contentLen = Record.getInt16(packet);          // pos: 3, 4
+
```

Comment:
Do you think it may be a good idea to check further the content of the alert including the first byte (WARNING or FATAL) and the second byte (AlertDescription for close_notify)?

Reply:
I'm not sure why. We only check that it's an alert type and our code downstream (`context.dispatch(plainText);`) actually processes the alert as needed.